app/src/main/java/com/ivanpodlesnykh/playlistmaker/ui/media/view_model/CreatePlaylistViewModel.kt

```diff
@@ -0,0 +1,57 @@
+package com.ivanpodlesnykh.playlistmaker.ui.media.view_model
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.ivanpodlesnykh.playlistmaker.domain.media.api.PlaylistInteractor
+import com.ivanpodlesnykh.playlistmaker.domain.media.models.Playlist
+import kotlinx.coroutines.launch
+
+class CreatePlaylistViewModel(
+    private val playlistInteractor: PlaylistInteractor
+) : ViewModel() {
+
+    private val imageUri = MutableLiveData<String>()
+    fun saveImageUri(uri: String) {
+        imageUri.value = uri
+    }
+
+    private val imageIsSetLiveData = MutableLiveData(false)
+    fun getImageIsSetLiveData(): LiveData<Boolean> = imageIsSetLiveData
+
+    private val titleIsSetLiveData = MutableLiveData(false)
+    fun getTitleIsSetLiveData(): LiveData<Boolean> = titleIsSetLiveData
+
+    private val descriptionIsSetLiveData = MutableLiveData(false)
+    fun getDescriptionIsSetLiveData(): LiveData<Boolean> = descriptionIsSetLiveData
+
+    fun setImage(isSet: Boolean) {
+        imageIsSetLiveData.value = isSet
+    }
+
+    fun setTitle(isSet: Boolean) {
+        titleIsSetLiveData.value = isSet
+    }
+
+    fun setDescription(isSet: Boolean) {
+        descriptionIsSetLiveData.value = isSet
+    }
+
+    fun createPlaylist(title: String, description: String) {
+
+        var savedUri = ""
+
+        if(imageUri.value.isNullOrEmpty()) savedUri = ""
+        else savedUri = playlistInteractor.saveImageToPrivateStorage(imageUri.value!!)
```

Comment:
```suggestion
        val savedUri = if(imageUri.value.isNullOrEmpty()) savedUri = ""
            else savedUri = playlistInteractor.saveImageToPrivateStorage(imageUri.value!!)
```
Можно чуть красивее написать, без `var`